theme/padplus/renderers.php

```diff
@@ -253,6 +253,93 @@ public function context_header($headerinfo = null, $headinglevel = 1) {
+                $showcoursemenu = true;
+            } else if (!empty($activenode) && ($activenode->type == navigation_node::TYPE_ACTIVITY ||
+                            $activenode->type == navigation_node::TYPE_RESOURCE)) {
+
+                // We only want to show the menu on the first page of the activity. This means
+                // the breadcrumb has no additional nodes.
+                if ($currentnode && ($currentnode->key == $activenode->key && $currentnode->type == $activenode->type)) {
+                    $showcoursemenu = true;
+                }
+            }
+        }
+
+        // This is the site front page.
+        if ($context->contextlevel == CONTEXT_COURSE &&
+                !empty($currentnode) &&
+                $currentnode->key === 'home') {
+            $showfrontpagemenu = true;
+        }
+
+        // This is the user profile page.
+        if ($context->contextlevel == CONTEXT_USER &&
+                !empty($currentnode) &&
+                ($currentnode->key === 'myprofile')) {
+            $showusermenu = true;
+        }
+
+        if ($showfrontpagemenu) {
+            // Allows to check the user permission for settings.
+            $settingsnode = $this->page->settingsnav->find('frontpage', navigation_node::TYPE_SETTING);
+            if ($settingsnode) {
+                $text = get_string('settings', 'theme_padplus');
+                $moodleurl = new moodle_url('/course/admin.php', array('courseid' => $this->page->course->id));
+                $url = "window.location.href='" . $moodleurl . "'";
+                $settingsbtn = html_writer::tag('button', $text, array('class' => 'btn btn-secondary', 'onclick' => $url));
+                return $settingsbtn;
+            }
+        } else if ($showcoursemenu) {
+            $settingsnode = $this->page->settingsnav->find('courseadmin', navigation_node::TYPE_COURSE);
+            if ($settingsnode) {
+                $text = get_string('settings', 'theme_padplus');
+                $moodleurl = new moodle_url('/course/admin.php', array('courseid' => $this->page->course->id));
+                $url = "window.location.href='" . $moodleurl . "'";
+                $settingsbtn = html_writer::tag('button', $text, array('class' => 'btn btn-secondary', 'onclick' => $url));
+                return $settingsbtn;
+            }
+        } else if ($showusermenu) {
+            $settingsnode = $this->page->settingsnav->find('useraccount', navigation_node::TYPE_CONTAINER);
+            if ($settingsnode) {
+                $text = get_string('settings', 'theme_padplus');
+                $moodleurl = new moodle_url('/user/preferences.php');
+                $url = "window.location.href='" . $moodleurl . "'";
+                $settingsbtn = html_writer::tag('button', $text, array('class' => 'btn btn-secondary', 'onclick' => $url));
+                return $settingsbtn;
+            }
+        }
+
+        return $this->render($menu);
```

Comment:
feel safe to replace this with the generated settings button. Or maybe an action_link ?